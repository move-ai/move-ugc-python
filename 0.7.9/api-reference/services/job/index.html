
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../take/">
      
      
        <link rel="next" href="../../schemas/file/">
      
      <link rel="icon" href="../../../assets/images/favicon.svg">
      <meta name="generator" content="mkdocs-1.5.1, mkdocs-material-9.1.18">
    
    
      
        <title>Jobs - Move UGC Python SDK</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.26e3688c.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.ecc896b0.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../../assets/_mkdocstrings.css">
    
    <script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="black" data-md-color-accent="black">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#jobs" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
        <aside class="md-banner">
          <div class="md-banner__inner md-grid md-typeset">
            
              <button class="md-banner__button md-icon" aria-label="Don't show this again">
                <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
              </button>
            
            
  <!-- keep length under 164 characters (less HTML tags) to fit on 1280px desktop window -->
This SDK is still under development and some docs may not reflect latest changes ðŸš§

          </div>
          
            <script>var content,el=document.querySelector("[data-md-component=announce]");el&&(content=el.querySelector(".md-typeset"),__md_hash(content.innerHTML)===__md_get("__announce")&&(el.hidden=!0))</script>
          
        </aside>
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
      </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../.." title="Move UGC Python SDK" class="md-header__button md-logo" aria-label="Move UGC Python SDK" data-md-component="logo">
      
  <img src="../../../assets/images/logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Move UGC Python SDK
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Jobs
            
          </span>
        </div>
      </div>
    </div>
    
      
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../.." class="md-tabs__link">
        Getting started
      </a>
    </li>
  

      
        
  
  
    
  


  
  
  
    

  
  
  
    <li class="md-tabs__item">
      <a href="../file/" class="md-tabs__link md-tabs__link--active">
        API Reference
      </a>
    </li>
  

  

      
        
  
  


  <li class="md-tabs__item">
    <a href="../../../contributing/" class="md-tabs__link">
      Contributing
    </a>
  </li>

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../.." title="Move UGC Python SDK" class="md-nav__button md-logo" aria-label="Move UGC Python SDK" data-md-component="logo">
      
  <img src="../../../assets/images/logo.svg" alt="logo">

    </a>
    Move UGC Python SDK
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1" >
      
      
      
        <label class="md-nav__link" for="__nav_1" id="__nav_1_label" tabindex="0">
          Getting started
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_1_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1">
          <span class="md-nav__icon md-icon"></span>
          Getting started
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../.." class="md-nav__link">
        Overview
      </a>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_1_2" >
      
      
      
        <label class="md-nav__link" for="__nav_1_2" id="__nav_1_2_label" tabindex="0">
          Usage
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_1_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_1_2">
          <span class="md-nav__icon md-icon"></span>
          Usage
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../getting-started/usage/file/" class="md-nav__link">
        Files
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../getting-started/usage/take/" class="md-nav__link">
        Takes
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../getting-started/usage/job/" class="md-nav__link">
        Jobs
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" checked>
      
      
      
        <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
          API Reference
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_2">
          <span class="md-nav__icon md-icon"></span>
          API Reference
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_1" checked>
      
      
      
        <label class="md-nav__link" for="__nav_2_1" id="__nav_2_1_label" tabindex="0">
          Services
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_1_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_2_1">
          <span class="md-nav__icon md-icon"></span>
          Services
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../file/" class="md-nav__link">
        Files
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../take/" class="md-nav__link">
        Takes
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Jobs
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Jobs
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#move_ugc.services.job.JobService" class="md-nav__link">
    JobService
  </a>
  
    <nav class="md-nav" aria-label="JobService">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#move_ugc.services.job.JobService.create" class="md-nav__link">
    create()
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#move_ugc.services.job.JobService.retrieve" class="md-nav__link">
    retrieve()
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2_2" >
      
      
      
        <label class="md-nav__link" for="__nav_2_2" id="__nav_2_2_label" tabindex="0">
          Schemas
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_2_2_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_2_2">
          <span class="md-nav__icon md-icon"></span>
          Schemas
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../schemas/file/" class="md-nav__link">
        File
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../schemas/client/" class="md-nav__link">
        Client
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../schemas/take/" class="md-nav__link">
        Take
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../schemas/job/" class="md-nav__link">
        Job
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../../contributing/" class="md-nav__link">
        Contributing
      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#move_ugc.services.job.JobService" class="md-nav__link">
    JobService
  </a>
  
    <nav class="md-nav" aria-label="JobService">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#move_ugc.services.job.JobService.create" class="md-nav__link">
    create()
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#move_ugc.services.job.JobService.retrieve" class="md-nav__link">
    retrieve()
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


<h1 id="jobs">Jobs</h1>


<div class="doc doc-object doc-class">



<h2 id="move_ugc.services.job.JobService" class="doc doc-heading">
        <span class="doc doc-object-name doc-class-name">JobService</span>


</h2>


  <div class="doc doc-contents first">

  
      <p>Service which can be used to communicate with Job type in UGC API.</p>
<p>To use this service, you need to instantiate it with a valid Move UGC client.</p>
<pre><code class="language-python">from move_ugc import MoveUgc
ugc = MoveUgc(api_key=&quot;my-api-key&quot;)

# Call job service methods directly
ugc.jobs.retrieve(id=&quot;job-&lt;guid&gt;&quot;)

# Or use the job service as a mixin
job_service = ugc.jobs
job_service.retrieve(id=&quot;job-&lt;guid&gt;&quot;)
</code></pre>



  

  <div class="doc doc-children">









<div class="doc doc-object doc-function">



<h3 id="move_ugc.services.job.JobService.create" class="doc doc-heading">
        <span class="doc doc-object-name doc-function-name">create</span>


</h3>
<div class="highlight"><pre><span></span><code><span class="n">create</span><span class="p">(</span><span class="n">take_id</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">metadata</span><span class="p">:</span> <span class="n">Optional</span><span class="p">[</span><span class="nb">str</span><span class="p">]</span> <span class="o">=</span> <span class="kc">None</span><span class="p">,</span> <span class="n">expand</span><span class="p">:</span> <span class="n">Optional</span><span class="p">[</span><span class="n">List</span><span class="p">[</span><span class="n">ALLOWED_EXPAND_ATTRS</span><span class="p">]]</span> <span class="o">=</span> <span class="kc">None</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">JobType</span>
</code></pre></div>

  <div class="doc doc-contents ">
  
      <p>Create a job in MoveUGC.</p>

  <p><strong>Parameters:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Name</th>
        <th>Type</th>
        <th>Description</th>
        <th>Default</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td><code>take_id</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>id of the take to be used for creating the job.</p>
            </div>
          </td>
          <td>
              <em>required</em>
          </td>
        </tr>
        <tr>
          <td><code>metadata</code></td>
          <td>
                <code><span title="typing.Optional">Optional</span>[str]</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>metadata to be used for creating the job. This should be a valid json string.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
        <tr>
          <td><code>expand</code></td>
          <td>
                <code><span title="typing.Optional">Optional</span>[<span title="typing.List">List</span>[<span title="move_ugc.schemas.constants.ALLOWED_EXPAND_ATTRS">ALLOWED_EXPAND_ATTRS</span>]]</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>list of fields to be expanded.
Currently only <code>client</code>, <code>take</code> and <code>outputs</code> are supported.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
    </tbody>
  </table>

  <p><strong>Returns:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Type</th>
        <th>Description</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td>
                <code><a class="autorefs autorefs-internal" title="move_ugc.schemas.job.JobType" href="../../schemas/job/#move_ugc.schemas.job.JobType">JobType</a></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Job instance of Pydantic model type.</p>
            </div>
          </td>
        </tr>
    </tbody>
  </table>

  </div>

</div>

<div class="doc doc-object doc-function">



<h3 id="move_ugc.services.job.JobService.retrieve" class="doc doc-heading">
        <span class="doc doc-object-name doc-function-name">retrieve</span>


</h3>
<div class="highlight"><pre><span></span><code><span class="n">retrieve</span><span class="p">(</span><span class="nb">id</span><span class="p">:</span> <span class="nb">str</span><span class="p">,</span> <span class="n">expand</span><span class="p">:</span> <span class="n">Optional</span><span class="p">[</span><span class="n">List</span><span class="p">[</span><span class="n">ALLOWED_EXPAND_ATTRS</span><span class="p">]]</span> <span class="o">=</span> <span class="kc">None</span><span class="p">)</span> <span class="o">-&gt;</span> <span class="n">JobType</span>
</code></pre></div>

  <div class="doc doc-contents ">
  
      <p>Retrieve a job from MoveUGC.</p>

  <p><strong>Parameters:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Name</th>
        <th>Type</th>
        <th>Description</th>
        <th>Default</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td><code>id</code></td>
          <td>
                <code>str</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>id of the job to be retrieved.</p>
            </div>
          </td>
          <td>
              <em>required</em>
          </td>
        </tr>
        <tr>
          <td><code>expand</code></td>
          <td>
                <code><span title="typing.Optional">Optional</span>[<span title="typing.List">List</span>[<span title="move_ugc.schemas.constants.ALLOWED_EXPAND_ATTRS">ALLOWED_EXPAND_ATTRS</span>]]</code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>list of fields to be expanded.
Currently only <code>client</code>, <code>take</code> and <code>outputs</code> are supported.</p>
            </div>
          </td>
          <td>
                <code>None</code>
          </td>
        </tr>
    </tbody>
  </table>

  <p><strong>Returns:</strong></p>
  <table>
    <thead>
      <tr>
        <th>Type</th>
        <th>Description</th>
      </tr>
    </thead>
    <tbody>
        <tr>
          <td>
                <code><a class="autorefs autorefs-internal" title="move_ugc.schemas.job.JobType" href="../../schemas/job/#move_ugc.schemas.job.JobType">JobType</a></code>
          </td>
          <td>
            <div class="doc-md-description">
              <p>Job instance of Pydantic model type.</p>
            </div>
          </td>
        </tr>
    </tbody>
  </table>

  </div>

</div>



  </div>

  </div>

</div>





                
              </article>
            </div>
          
          
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </button>
        
      </main>
      
        <footer class="md-footer">
  
    
      
      <nav class="md-footer__inner md-grid" aria-label="Footer" >
        
          
          <a href="../take/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Takes" rel="prev">
            <div class="md-footer__button md-icon">
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
            </div>
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Previous
              </span>
              <div class="md-ellipsis">
                Takes
              </div>
            </div>
          </a>
        
        
          
          <a href="../../schemas/file/" class="md-footer__link md-footer__link--next" aria-label="Next: File" rel="next">
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Next
              </span>
              <div class="md-ellipsis">
                File
              </div>
            </div>
            <div class="md-footer__button md-icon">
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg>
            </div>
          </a>
        
      </nav>
    
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      <a href='https://www.move.ai/' target=_blank>&copy; Move Ai Limited</a>
    </div>
  
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../../..", "features": ["announce.dismiss", "navigation.tabs", "navigation.footer", "navigation.instant", "navigation.tabs.sticky", "navigation.top", "content.code.copy", "content.code.annotate"], "search": "../../../assets/javascripts/workers/search.74e28a9f.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"provider": "mike"}}</script>
    
    
      <script src="../../../assets/javascripts/bundle.220ee61c.min.js"></script>
      
    
  </body>
</html>