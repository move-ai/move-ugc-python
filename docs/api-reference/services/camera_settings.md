# Camera settings

::: move_ugc.services.camera_settings.CameraSettingsService
    options:
        members:
            - list
