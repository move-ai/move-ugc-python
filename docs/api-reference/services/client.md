# Client

::: move_ugc.services.client.ClientService
    options:
        members:
            - retrieve
            - update