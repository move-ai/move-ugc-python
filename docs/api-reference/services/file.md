# Files

::: move_ugc.services.file.FileService
    options:
        members:
            - create
            - retrieve