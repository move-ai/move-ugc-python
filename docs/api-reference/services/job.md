# Jobs

::: move_ugc.services.job.JobService
    options:
        members:
            - create
            - retrieve
            - update
            - list
