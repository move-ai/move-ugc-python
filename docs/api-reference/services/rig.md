# Rig

::: move_ugc.services.rig.RigService
    options:
        members:
            - list
