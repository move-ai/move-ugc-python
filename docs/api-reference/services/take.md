# Takes

::: move_ugc.services.take.TakeService
    options:
        members:
            - create_singlecam
            - create_multicam
            - retrieve
            - list
            - update