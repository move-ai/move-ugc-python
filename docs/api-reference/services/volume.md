# Volume

::: move_ugc.services.take.VolumeService
    options:
        members:
            - create_human_volume
            - retrieve_human_volume