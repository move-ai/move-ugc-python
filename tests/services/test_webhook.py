"""Unit tests for using the webhook service."""

from tests.services.testcases import ServicesTestCase


class TestWebhooksService(ServicesTestCase):
    """Test webhook service."""

    service_name = "webhooks"

    def test_add_webhook(
        self,
        faker,
        upsert_webhook_endpoint_response,
        snapshot,
        snapshot_json,
    ):
        """Test adding a webhook endpoint to Move UGC API.

        Args:
            faker: The faker fixture.
            upsert_webhook_endpoint_response: mocked webhook endpoint response
            snapshot: The snapshot fixture.
            snapshot_json: The snapshot json fixture.
        """
        webhook_model = self.client.webhooks.upsert(
            events=["ugc.jobs.state.completed"],
            uid=faker.pystr(),
            url=faker.url(),
            secret=faker.pystr(),
            description=faker.pystr(),
        )
        self.assert_execute(snapshot, name="add_webhook_query")
        assert webhook_model.model_dump() == snapshot_json(
            name="add_webhook_response",
        )

    def test_add_without_secret(
        self,
        faker,
        upsert_webhook_endpoint_response,
        snapshot,
        snapshot_json,
    ):
        """Test adding a webhook endpoint to Move UGC API without pre-defined secret.

        Args:
            faker: The faker fixture.
            upsert_webhook_endpoint_response: mocked webhook endpoint response
            snapshot: The snapshot fixture.
            snapshot_json: The snapshot json fixture.
        """
        webhook_model = self.client.webhooks.upsert(
            events=["ugc.jobs.state.completed"],
            uid=faker.pystr(),
            url=faker.url(),
        )
        self.assert_execute(snapshot, name="add_webhook_query")
        assert webhook_model.model_dump() == snapshot_json(
            name="add_webhook_response",
        )
