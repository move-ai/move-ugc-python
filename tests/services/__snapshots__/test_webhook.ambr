# serializer version: 1
# name: TestWebhooksService.test_add_webhook[add_webhook_query]
  list([
    list([
      '''
        mutation addWebhook($events: [String!]!, $uid: String!, $url: String!, $description: String, $secret: String) {
          upsertWebhookEndpoint(
            events: $events
            uid: $uid
            url: $url
            description: $description
            secret: $secret
          ) {
            description
            events
            uid
            url
            secret
          }
        }
      ''',
    ]),
    dict({
      'operation_name': None,
      'variable_values': dict({
        'description': 'ZPnbQcVNCliYtuFCSJkG',
        'events': list([
          'ugc.jobs.state.completed',
        ]),
        'secret': 'XbBPNrbhtJksbBuoWXSK',
        'uid': 'WkDQwclzAvJMvacZIYSm',
        'url': 'https://dunn.com/',
      }),
    }),
  ])
# ---
# name: TestWebhooksService.test_add_without_secret[add_webhook_query]
  list([
    list([
      '''
        mutation addWebhook($events: [String!]!, $uid: String!, $url: String!, $description: String, $secret: String) {
          upsertWebhookEndpoint(
            events: $events
            uid: $uid
            url: $url
            description: $description
            secret: $secret
          ) {
            description
            events
            uid
            url
            secret
          }
        }
      ''',
    ]),
    dict({
      'operation_name': None,
      'variable_values': dict({
        'description': '',
        'events': list([
          'ugc.jobs.state.completed',
        ]),
        'secret': None,
        'uid': 'WkDQwclzAvJMvacZIYSm',
        'url': 'https://dunn.com/',
      }),
    }),
  ])
# ---
# name: TestWebhooksService.test_fetch_service
  WebhookService(api_key=SecretStr('**********'), endpoint_url=HttpUrl('https://pytest_invalid_endpoint_url.com/'))
# ---
